src/circular_buffer.jl

```diff
@@ -1,22 +1,38 @@
 """
-    CircularBuffer{T}(n)
+    CircularBuffer{T}(v,n::Int)
 
 The CircularBuffer type implements a circular buffer of fixed capacity
 where new items are pushed to the back of the list, overwriting values
 in a circular fashion.
 
-Allocate a buffer of elements of type `T` with maximum capacity `n`.
+Allocate a buffer of elements of type `T`  containing items `v` with maximum capacity `n`.
+If no capacity is provided the capacity is the number of elements in `v`
 """
 mutable struct CircularBuffer{T} <: AbstractVector{T}
     capacity::Int
     first::Int
     length::Int
     buffer::Vector{T}
 
-    CircularBuffer{T}(capacity::Int) where {T} = new{T}(capacity, 1, 0, Vector{T}(undef, capacity))
```

Comment:
Possibly we still want an inner constructor her to check that 
`first <= length <= capacity` and that `capacity==length(buffer)` (possibly capacity shouldn't be passed in here now, just `first`, `length`, and `buffer` ?

Reply:
Apologies for the delay, work and the holiday weekend have kept me busy. I hope to get this wrapped up this week. Thanks for the mentorship throughout this process.

I agree that the "internal" interface should be passing the vector, the first index, and the length, but skipping the capacity. There isn't anything useful one can do by passing a capacity different from length(vector). Whereas atleast conceivably by carefully constructing a vector and setting first/length correctly you could directly create an instance of a buffer with any internal structure that you could arrive at via push/pop. I'll implement that internal constructor with the checks, and make the few other edits needed to the other constructors to match that new format in the next few days.

Reply:
cool thanks, sorry for my own slowness